src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -241,6 +241,12 @@ public override void Execute(DocumentsOperationContext context)
                             cmd.Document.TryGet(Constants.Documents.Metadata.Key, out metadata);
                             LastEtag = putResult.Etag;
 
+                        
+                            metadata.Modifications = new DynamicJsonValue(metadata)
```

Comment:
Basically, the issue is that we lose the Etag & Id in the metadata on the return call.
This is bad for the client, because it needs to stitch them together, since we are writing it back out again, we can just put it there at very little additional cost.